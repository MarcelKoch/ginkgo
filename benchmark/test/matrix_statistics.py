#!/usr/bin/env python3
import test_framework
# check that all input modes work:
# parameter
test_framework.compare_output(["matrix_statistics/matrix_statistics", "-input", '[{"size": 100, "stencil": "7pt"}]'],
                              expected_stdout="matrix_statistics.simple.stdout",
                              expected_stderr="matrix_statistics.simple.stderr")

# stdin
test_framework.compare_output(["matrix_statistics/matrix_statistics"],
                              expected_stdout="matrix_statistics.simple.stdout",
                              expected_stderr="matrix_statistics.simple.stderr",
                              stdin='[{"size": 100, "stencil": "7pt"}]')

# input file
test_framework.compare_output(["matrix_statistics/matrix_statistics", "-input", '[{"size": 100, "stencil": "7pt"}]'],
                              expected_stdout="matrix_statistics.simple.stdout",
                              expected_stderr="matrix_statistics.simple.stderr")
