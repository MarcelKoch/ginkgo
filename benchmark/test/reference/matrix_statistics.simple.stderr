This is Ginkgo 1.5.0 (develop)
    running with core module 1.5.0 (develop)
Running test case
{
    "size": 100,
    "stencil": "7pt",
    "problem": {}
}
Matrix is of size (125, 125)
